atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -62,12 +62,17 @@ public LockWatchValueScopingCacheImpl(
     public synchronized void processStartTransactions(Set<Long> startTimestamps) {
         TransactionsLockWatchUpdate updateForTransactions =
                 eventCache.getUpdateForTransactions(startTimestamps, currentVersion);
+
+        if (updateForTransactions.clearCache()) {
+            clearCache();
+        }
```

Comment:
Oops! I think I was technically doing this by calculating it myself, except that I was doing it _after_ updating the stores, which means that I was making it worse for myself.